lib/msal-core/src/utils/UrlUtils.ts

```diff
@@ -244,4 +245,27 @@ export class UrlUtils {
         const hash = UrlUtils.getHashFromUrl(urlFragment);
         return CryptoUtils.deserialize(hash);
     }
+
+    /**
+     * @hidden
+     * generate unique state per request
+     * @param request
+     */
+    static getRequestState(state: string): string {
+        // append GUID to user set state  or set one for the user if null
+        return state && !StringUtils.isEmpty(state) ? CryptoUtils.createNewGuid() + "|" + state : CryptoUtils.createNewGuid();
+    }
+
+    /**
+     * @hidden
+     * validate correlationId and generate if not valid or not set by the user
+     * @param correlationId
+     */
+    static getRequestCorrelationId(correlationId: string): string {
+        // validate user set correlationId or set one for the user if null
+        if(correlationId && !CryptoUtils.isGuid(correlationId)) {
+            throw ClientConfigurationError.createInvalidCorrelationIdError();
+        }
+        return correlationId && CryptoUtils.isGuid(correlationId)? correlationId : CryptoUtils.createNewGuid();
```

Comment:
Don't need to truthy check `correlationId`, since you are doing that above.

Reply:
Changes made.